### ISSUE TITLE
arrow functions parameters scope

### ISSUE THREAD

**Issue by mistadikay** [babel/babel-eslint#13]
```js
something(e) {
  ...
  console.log(e.target);

  reader.addEventListener('load', e => {
    console.log(e.target);
  }, false);

  ...
}
```
In the example above `e` parameter in arrow function belongs to arrow function scope and not interfering with `e` parameter in the upper scope. But I'm having an error:
```bash
25:45  error    e is already declared in the upper scope  no-shadow
```

**Comment by sebmck** [comment]
Yes, that's exactly what the `no-shadow` rule does and it's working as expected. From the ESLint docs:

> Shadowing is the process by which a local variable shares the same name as a variable in its containing scope.

http://eslint.org/docs/rules/no-shadow.html

You can disable it via `.eslintrc`:

```json
{
  "parser": "babel-eslint",
  "rules": {
    "no-shadow": 0
  }
}
```

**Comment by mistadikay** [comment]
I see, my bad. Thanks!
